elastic/kibana
# [Uptime] Add migration to include synthetics and heartbeat indices on 8.2.0

## Summary

This PR adds a migration to include Synthetics and Heartbeat indices on 8.2.0 and thus fix problems for customers which had `heartbeat-7*` indices before (and didn't have `synthetics-*`).

For that, I've followed the guide [here](https://docs.elastic.dev/kibana-dev-docs/tutorials/testing-plugins#saved-objects-migrations) and have spoken to @shahzad31 too about how to go about testing it.

Fixes https://github.com/elastic/kibana/issues/129297


## How to test this PR

1. Using a branch _without_ the change, start Kibana and ES
    TIP: You can do that using the latest snapshot version and `elastic-package stack up`.
2. Log into Kibana, go to Uptime > Settings, and change the indices used to find heartbeat documents so that they don't contain either `heartbeat-8*` nor `synthetics-*`.
3. Checkout this branch and start Kibana locally with these changes.
4. Check your indices settings again and you should see both `heartbeat-8*` and `synthetics-*`, plus any other indices that were already there.


### Checklist

Delete any items that are not applicable to this PR.


(Only the one above seems to apply)


### For maintainers
